src/Main.elm

```diff
@@ -0,0 +1,87 @@
+module Main exposing (init, main, update, view)
+
+import Browser
+import Element exposing (centerX, centerY, el, layout, map, text)
+import Element.Input exposing (button)
+import Ranking
+import Setup
+
+
+main : Program () Model Msg
+main =
+    Browser.sandbox
+        { init = init
+        , view = layout [] << view
+        , update = update
+        }
+
+
+type Msg
+    = SetupMsg Setup.Msg
+    | RankingMsg Ranking.Msg
+    | SwitchView
+
+
+type alias Model =
+    { setup : Setup.Model
+    , ranking : Ranking.Model
+    , screen : CurrentScreen
+    }
+
+
+type CurrentScreen
+    = Setup
+    | Ranking
+
+
+init : Model
+init =
+    { setup = Setup.init
+    , ranking = Ranking.init
+    , screen = Setup
+    }
+
+
+update : Msg -> Model -> Model
+update msg model =
+    case msg of
+        SetupMsg setupMsg ->
+            { model | setup = Setup.update setupMsg model.setup }
+
+        RankingMsg rankingMsg ->
+            { model | ranking = Ranking.update rankingMsg model.ranking }
+
+        SwitchView ->
+            { model
+                | screen =
+                    case model.screen of
+                        Setup ->
+                            Ranking
+
+                        Ranking ->
+                            Setup
+            }
+
+
+view : Model -> Element.Element Msg
+view model =
+    let
+        switchScreen =
+            button []
+                { onPress = Just SwitchView
+                , label = text "switch view"
+                }
+
+        element xMsg xView xModel =
+            el [ centerX, centerY ]
+                << map xMsg
+                << xView
+            <|
+                xModel model
```

Comment:
Nice :slightly_smiling_face: 
You should give this more appropriate names though.
I'd also add type annotations and replace the compositions with normal applications (pipes), because the advantage of a composition is enabling point-free definitions.

How about:
```elm
centerred : (innerMsg -> msg) -> (innerModel -> Element.Element innerMsg) -> (model -> innerModel) -> Element.Element msg
centerred wrap viewInner extract =
    extract model
        |> viewInner
        |> el [ centerX, centerY ]
        |> map wrap
```

This can be used anywhere, so it shouldn't be hidden inside `view`.
If you put this in a helpers module (like `Colors.elm`) it can save boilerplate all over the project.